# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: js/src/vm/ObjectGroup.h
# Commit: 4531d5c25694
# Full Hash: 4531d5c256947ff782377ba0143e3f37bd7207c2
# Author: Jan de Mooij <jdemooij@mozilla.com>
# Date: 2021-03-06 09:33:16
# Regressor Bug: 1689413
# File Overlap Count: 1
# Description:
#   Bug 1689413 part 5 - Remove JSClass* from ObjectGroup. r=jonco
#   
#   BaseShape stores the JSClass* too so this patch changes the code to get the JSClass*
#   from there instead of from the ObjectGroup.
#   
# ==============================================================================

diff -r f1bb16079c0e -r 4531d5c25694 js/src/vm/ObjectGroup.h
--- a/js/src/vm/ObjectGroup.h	Fri Mar 05 19:10:06 2021 +0000
+++ b/js/src/vm/ObjectGroup.h	Fri Mar 05 19:10:06 2021 +0000
@@ -42,12 +42,10 @@
   TenuredObject
 };
 
-class ObjectGroup : public gc::TenuredCellWithNonGCPointer<const JSClass> {
- public:
-  /* Class shared by objects in this group, stored in the cell header. */
-  const JSClass* clasp() const { return headerPtr(); }
-
- private:
+// Note: for now just store an int* in the CellHeader. We can't store the Realm*
+// because it's an incomplete type and we can't include vm/Realm.h here. This
+// class will be removed shortly anyway.
+class ObjectGroup : public gc::TenuredCellWithNonGCPointer<int> {
   /* Prototype shared by objects in this group. */
   GCPtr<TaggedProto> proto_;  // set by constructor
 
@@ -57,8 +55,6 @@
   // END OF PROPERTIES
 
  private:
-  static inline uint32_t offsetOfClasp() { return offsetOfHeaderPtr(); }
-
   static inline uint32_t offsetOfProto() {
     return offsetof(ObjectGroup, proto_);
   }
@@ -73,7 +69,7 @@
   friend class js::jit::MacroAssembler;
 
  public:
-  inline ObjectGroup(const JSClass* clasp, TaggedProto proto, JS::Realm* realm);
+  inline ObjectGroup(TaggedProto proto, JS::Realm* realm);
 
   const GCPtr<TaggedProto>& proto() const { return proto_; }
 